api/tests/AppBundle/Controller/CasRecControllerTest.php

```diff
@@ -54,37 +54,43 @@ public function setUp(): void
         ]);
     }
 
-    private function compress($data)
+    public function testDeleteBySourceVerifiesSourceInput()
     {
-        return base64_encode(gzcompress(json_encode($data), 9));
-    }
+        $this->buildAndPersistCasRecEntity('12345678', CasRec::CASREC_SOURCE);
+        $this->fixtures()->flush();
+        $this->fixtures()->clear();
 
-    public function testTruncate()
-    {
-        // just to check it gets truncated
-        $casRec = new CasRec([
-            'Case' => 'case',
-            'Surname' => 'I should get deleted',
-            'Deputy No' => 'Deputy No',
-            'Dep Surname' => 'Dep Surname',
-            'Dep Postcode' => 'SW1',
-            'Typeofrep'=>'OPG102',
-            'Corref'=>'L2'
+        $url = '/casrec/delete-by-source/unknownsource';
+
+        $this->assertJsonRequest('DELETE', $url, [
+            'mustSucceed' => false,
```

Comment:
I think this should be:
```suggestion
            'mustFail' => true,
```
If `mustSucceed` is false, it won't check anything